### DISCUSSION TITLE
AngularLS crashes with `ngtypecheck not found` after saving

### DISCUSSION THREAD

**Discussion by l3d00m** [LazyVim/LazyVim#3970] [General]
The original issue #3879 was closed as not being caused by LazyVim, so I'm opening this discussion in hopes of finding more people with this issue and also workarounds for it. 

### Related
- Closed issue #3879 
- Issue in different repo: https://github.com/angular/vscode-ng-language-service/issues/2043
- [Stackoverflow question](https://vi.stackexchange.com/questions/45092/angular-language-server-crashes-when-working-on-an-nx-project-with-nx-serve)

### Steps To Reproduce

This issue occurs more often under Windows, but I've also had it happen with Linux.

1. Setup a clean new angular v18 project using `npm install -g @angular/cli` followed by `ng new my-app` with the default settings (just press enter).
2. Start nvim with the config below: `nvim -u repro.lua` (it's stock lazyvim+ [angular LSP extra](https://www.lazyvim.org/extras/lang/angular) enabled)
3. Open `app.component.ts `. 
4. Save the file a few times.
5. After a few saves (at most 3 for me), the LSP crashes with `Client angularls quit with exit code 1 and signal 0. Check log for errors: C:\Users\myuser\AppData\Local\nvim-data/lsp.log` and in the log it says  `Program does not contain "c:/Users/myuser/IdeaProjects/projects/my-app/org/apps/org/src/app/app.component.ngtypecheck.ts"`

### `repro.lua` config

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  { "LazyVim/LazyVim", import = "lazyvim.plugins" },
  { import = "lazyvim.plugins.extras.lang.angular" },
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

Probably relevant for @Elias-Graf and @zer09 as well. Also letting @radvil know since they introduced the angular lsp (thanks!).

**Comment by l3d00m** [comment]
To anyone else who has this happening: 

- Are you also using the LazyVim extra? (So far I couldn't reproduce it otherwise)
- Which OS are you using?
- Which angular version are you using?

**Comment by l3d00m** [comment]
I have also created the following config, where I've manually overwritten the config, similar to the angular extra. But with this config, the issue never occured to me and I'm not sure why.

```lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--branch=stable",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  { "LazyVim/LazyVim", import = "lazyvim.plugins" },
  { -- LSP Configuration & Plugins
    "neovim/nvim-lspconfig",
    dependencies = {
      -- Automatically install LSPs and related tools to stdpath for Neovim
      { "williamboman/mason.nvim", config = true }, -- NOTE: Must be loaded before dependants
      "williamboman/mason-lspconfig.nvim",
      "WhoIsSethDaniel/mason-tool-installer.nvim",
    },
    config = function()
      local capabilities = vim.lsp.protocol.make_client_capabilities()
      local servers = {
        vtsls = {
          tsserver = {
            globalPlugins = {
              name = "@angular/language-server",
              location = root .. "data/nvim-data/mason/bin",
              enableforworkspacetypescriptversions = false,
            },
          },
        },
        angularls = {},
      }

      require("mason").setup()

      local ensure_installed = vim.tbl_keys(servers or {})
      require("mason-tool-installer").setup({ ensure_installed = ensure_installed })

      require("mason-lspconfig").setup({
        handlers = {
          function(server_name)
            local server = servers[server_name] or {}
            server.capabilities = vim.tbl_deep_extend("force", {}, capabilities, server.capabilities or {})
            require("lspconfig")[server_name].setup(server)
          end,
        },
      })
    end,
  },
  -- add any other plugins here
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})


```

**Comment by zer09** [comment]
I got this error when `mason` updated the `angularls` to the latest version `v18`, as of now my workaround is to use `v17.3.2`.

```lua
{
	"williamboman/mason-lspconfig.nvim",
	opts = {
		ensure_installed = {
			"angularls@17.3.2",
		},
	},
},
```

I'm not using `LazyVim` and got same problem, so I think the cause of the problem is the latest release of `angularls`.

**Reply by l3d00m** [reply]
Thanks, that might help! Is your project based on angular v18 oder v17?

**Comment by Elias-Graf** [comment]
Not that insightful, but I've been experiencing this with "normal" nvim, lazy.nvim (plugin manager), and mason. So probaly not related to the LazyVim distribution.

**Reply by l3d00m** [reply]
That's good to know too, thanks. Then I'll try to  build a minimal reproduction example without lazyvim.

**Reply by Elias-Graf** [reply]
The app itself is ~ `17.2.4`, the language server is `18.0.0`. AFAIK it should be backwards compatible.

**Comment by radvil** [comment]
This only occured if i have older angular app, and use the latest version of the lsp, not related to lazyvim, u should downgrade the server to 17. 18 will always complain even in vscode.
